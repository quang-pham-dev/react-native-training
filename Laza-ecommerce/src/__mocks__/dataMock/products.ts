import { ProductProps } from 'types/Products';

export const product: ProductProps = {
  id: '0',
  brandId: '1',
  name: 'Nike',
  price: 100,
  type: 'shoes',
  like: true,
  title: 'Nike Sportswear Club Fleece',
  source:
    'https://images.unsplash.com/photo-1542291026-7eec264c27ff?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80',
  description:
    "Nike is a multinaelichick and retired quarterback Jack Nicklaus. Nike's products include apparel, shoes, equipment, and accessories. Nike also owns and operates the majority of the world's sporting goods brands, including Adidas, Puma, and Fila.",
  reviewer: {
    id: '2',
    name: 'John Doe',
    date: '13, Sep, 2020',
    image:
      'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
  },
  imageReview: [
    {
      id: '0',
      image:
        'https://images.unsplash.com/photo-1499852848443-3004d6dc4cfc?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1374&q=80',
    },
    {
      id: '1',
      image:
        'https://images.unsplash.com/photo-1618898909019-010e4e234c55?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80',
    },
    {
      id: '2',
      image:
        'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
    },
    {
      id: '3',
      image:
        'https://images.unsplash.com/photo-1518656306295-aa28b28b2504?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
    },
    {
      id: '4',
      image:
        'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
    },
    {
      id: '5',
      image:
        'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
    },
  ],
  rating: '4.5',
  comment:
    'lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,adipisicing elit. Quisquam, quia.',
};

export const products: ProductProps[] = [
  {
    id: '0',
    brandId: '1',
    name: 'Nike',
    price: 100,
    type: 'shoes',
    like: true,
    title: 'Nike Sportswear Club Fleece',
    source:
      'https://images.unsplash.com/photo-1542291026-7eec264c27ff?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80',
    description:
      "Nike is a multinaelichick and retired quarterback Jack Nicklaus. Nike's products include apparel, shoes, equipment, and accessories. Nike also owns and operates the majority of the world's sporting goods brands, including Adidas, Puma, and Fila.",
    reviewer: {
      id: '2',
      name: 'John Doe',
      date: '13, Sep, 2020',
      image:
        'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
    },
    imageReview: [
      {
        id: '0',
        image:
          'https://images.unsplash.com/photo-1499852848443-3004d6dc4cfc?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1374&q=80',
      },
      {
        id: '1',
        image:
          'https://images.unsplash.com/photo-1618898909019-010e4e234c55?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80',
      },
      {
        id: '2',
        image:
          'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
      {
        id: '3',
        image:
          'https://images.unsplash.com/photo-1518656306295-aa28b28b2504?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
      {
        id: '4',
        image:
          'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
      },
      {
        id: '5',
        image:
          'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
    ],
    rating: '4.5',
    comment:
      'lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,adipisicing elit. Quisquam, quia.',
  },
  {
    id: '0',
    brandId: '1',
    name: 'Nike',
    price: 100,
    type: 'shoes',
    like: true,
    title: 'Nike Sportswear Club Fleece',
    source:
      'https://images.unsplash.com/photo-1542291026-7eec264c27ff?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470&q=80',
    description:
      "Nike is a multinaelichick and retired quarterback Jack Nicklaus. Nike's products include apparel, shoes, equipment, and accessories. Nike also owns and operates the majority of the world's sporting goods brands, including Adidas, Puma, and Fila.",
    reviewer: {
      id: '2',
      name: 'John Doe',
      date: '13, Sep, 2020',
      image:
        'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
    },
    imageReview: [
      {
        id: '0',
        image:
          'https://images.unsplash.com/photo-1499852848443-3004d6dc4cfc?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1374&q=80',
      },
      {
        id: '1',
        image:
          'https://images.unsplash.com/photo-1618898909019-010e4e234c55?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=687&q=80',
      },
      {
        id: '2',
        image:
          'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
      {
        id: '3',
        image:
          'https://images.unsplash.com/photo-1518656306295-aa28b28b2504?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
      {
        id: '4',
        image:
          'https://images.unsplash.com/photo-1609505848912-b7c3b8b4beda?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=765&q=80',
      },
      {
        id: '5',
        image:
          'https://images.unsplash.com/photo-1575537302964-96cd47c06b1b?crop=entropy&cs=tinysrgb&fm=jpg&ixlib=rb-1.2.1&q=80&raw_url=true&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1470',
      },
    ],
    rating: '4.5',
    comment:
      'lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,adipisicing elit. Quisquam, quia.',
  },
];
