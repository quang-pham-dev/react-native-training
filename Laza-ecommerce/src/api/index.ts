import { authService } from './auth.api';
import http from './http';

export { http, authService };
