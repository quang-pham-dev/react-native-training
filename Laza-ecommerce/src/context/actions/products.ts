import { IProduct } from 'types/models/Products';

export const GET_PRODUCTS = 'GET_PRODUCTS';
export const GET_PRODUCTS_SUCCESS = 'GET_PRODUCTS_SUCCESS';
export const GET_PRODUCTS_FAILED = 'GET_PRODUCTS_FAILED';

export const GET_PRODUCT = 'GET_PRODUCT';
export const GET_PRODUCT_SUCCESS = 'GET_PRODUCT_SUCCESS';
export const GET_PRODUCT_FAILED = 'GET_PRODUCT_FAILED';

export const GET_PRODUCTS_BY_BRAND_ID = 'GET_PRODUCTS_BY_BRAND_ID';
export const GET_PRODUCTS_BY_BRAND_ID_SUCCESS = 'GET_PRODUCTS_BY_BRAND_ID_SUCCESS';
export const GET_PRODUCTS_BY_BRAND_ID_FAILED = 'GET_PRODUCTS_BY_BRAND_ID_FAILED';

export const RESET_STATE = 'RESET_STATE';

export interface ProductsAction {
  payload?: IProduct[];
  type: string;
  error?: string;
}
