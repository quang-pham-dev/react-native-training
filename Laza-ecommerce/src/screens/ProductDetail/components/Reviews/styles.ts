import { StyleSheet } from 'react-native';

// Themes
import Colors from 'themes/Colors';
import Fonts from 'themes/Fonts';
import Metrics from 'themes/Metrics';

const styles = StyleSheet.create({
  reviewContainer: {
    marginTop: Metrics.margin.lg,
  },

  reviewTitleWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },

  textReviewTitle: {
    fontSize: Fonts.size.default,
    fontFamily: Fonts.fontFamily.Inter_600SemiBold,
    lineHeight: Fonts.lineHeight.sm,
    color: Colors.textBlack,
  },

  reviewWrapper: {
    width: '100%',
    justifyContent: 'space-between',
  },

  reviewItemWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginTop: Metrics.margin.xm,
  },

  reviewerInfoWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },

  imageAvatar: {
    width: 40,
    height: 40,
    borderRadius: 40,
  },

  reviewerItemInfo: {
    marginLeft: Metrics.margin.xm,
  },

  reviewTime: {
    flexDirection: 'row',
    alignItems: 'center',
    marginTop: Metrics.margin.xs,
  },

  imageClock: {
    width: Metrics.icons.small,
    height: Metrics.icons.small,
  },

  textDate: {
    fontSize: Fonts.size.min,
    fontFamily: Fonts.fontFamily.Inter_500Medium,
    lineHeight: Fonts.lineHeight.xxs,
    color: Colors.textGray,
    marginLeft: Metrics.margin.xsm,
  },

  textName: {
    fontSize: Fonts.size.normal,
    fontFamily: Fonts.fontFamily.Inter_500Medium,
    lineHeight: Fonts.lineHeight.base,
  },
  reviewItemRating: {
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },

  ratingTextWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },

  ratingTextPoint: {
    fontSize: Fonts.size.normal,
    fontFamily: Fonts.fontFamily.Inter_500Medium,
    lineHeight: Fonts.lineHeight.base,
    color: Colors.textBlack,
  },
  ratingText: {
    fontSize: Fonts.size.min,
    fontFamily: Fonts.fontFamily.Inter_400Regular,
    lineHeight: Fonts.lineHeight.xxs,
    color: Colors.textGray,
    marginLeft: Metrics.margin.xsm,
  },

  imageStar: {
    width: 57,
    height: 13,
  },
  reviewItemContent: {
    marginTop: Metrics.margin.sm,
  },
  textContent: {
    fontSize: Fonts.size.normal,
    fontFamily: Fonts.fontFamily.Inter_400Regular,
    lineHeight: Fonts.lineHeight.lg,
    color: Colors.textGray,
  },

  textTitle: {
    fontSize: Fonts.size.small,
    fontFamily: Fonts.fontFamily.Inter_400Regular,
    lineHeight: Fonts.lineHeight.xs,
    color: Colors.textGray,
  },
});

export default styles;
