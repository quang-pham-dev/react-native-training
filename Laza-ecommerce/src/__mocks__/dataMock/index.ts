import { user } from './user';
import { brand, brands } from './brands';
import { product, products } from './products';

export { brand, brands, product, products, user };
