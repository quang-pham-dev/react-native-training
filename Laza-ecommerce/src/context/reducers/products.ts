import {
  GET_PRODUCT,
  GET_PRODUCTS,
  GET_PRODUCT_SUCCESS,
  GET_PRODUCTS_SUCCESS,
  GET_PRODUCT_FAILED,
  GET_PRODUCTS_FAILED,
  GET_PRODUCTS_BY_BRAND_ID,
  GET_PRODUCTS_BY_BRAND_ID_SUCCESS,
  GET_PRODUCTS_BY_BRAND_ID_FAILED,
  RESET_STATE,
  ProductsAction,
} from 'context/actions/products';
import { InitialProductsState, ProductsState } from 'context/state/products';

const productsReducer = (state: ProductsState = InitialProductsState, action: ProductsAction) => {
  switch (action.type) {
    case GET_PRODUCTS:
    case GET_PRODUCT:
    case GET_PRODUCTS_BY_BRAND_ID:
      return {
        ...state,
        isProcessing: true,
      };

    case GET_PRODUCTS_SUCCESS:
      return {
        ...state,
        type: action.type,
        isProcessing: false,
        products: action.payload,
      };

    case GET_PRODUCT_SUCCESS:
      return {
        ...state,
        type: action.type,
        isProcessing: false,
        productById: action.payload,
      };

    case GET_PRODUCTS_BY_BRAND_ID_SUCCESS:
      return {
        ...state,
        type: action.type,
        isProcessing: false,
        productsByBrandId: action.payload,
      };

    case GET_PRODUCTS_FAILED:
    case GET_PRODUCT_FAILED:
    case GET_PRODUCTS_BY_BRAND_ID_FAILED:
      return {
        ...state,
        type: action.type,
        error: action.error,
      };

    case RESET_STATE:
      return {
        ...InitialProductsState,
      };

    default:
      /* If this reducer doesn't recognize the action type, or doesn't
     care about this specific action, return the existing state unchanged */
      throw new Error(`Unknown action type: ${action.type}`);
  }
};

export default productsReducer;
