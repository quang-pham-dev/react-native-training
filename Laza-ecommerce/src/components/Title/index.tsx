import React, { memo } from 'react';
import { Text } from 'react-native';

// LIBS
import isEqual from 'react-fast-compare';

// Types
import { ITitleProps } from 'types/components/Title';

// Styles
import styles from './styles';

const Title = ({ titleName, titleStyles }: ITitleProps) => {
  return <Text style={[styles.title, titleStyles]}>{titleName}</Text>;
};

export default memo(Title, isEqual);
