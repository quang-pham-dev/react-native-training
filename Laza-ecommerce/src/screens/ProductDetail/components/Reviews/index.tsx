import React, { memo } from 'react';
import { View, Text, Image } from 'react-native';

// LIBS
import isEqual from 'react-fast-compare';

// Components
import Title from 'components/Title';
import MoreLessText from 'components/LessMoreText';

// Types
import { IProductDetailItemProps } from 'types/screens/ProductDetail';

// Themes
import IMAGES from 'themes/Images';

// Styles
import styles from './styles';

const Reviews = ({ comment, rating, reviewers }: IProductDetailItemProps) => {
  return (
    <View style={styles.reviewContainer}>
      <View style={styles.reviewTitleWrapper}>
        <Title.titleLeftSection titleName='Reviews' />
        <Title.titleRightSection titleName='View All' />
      </View>
      {/* End of reviews Title */}

      <View style={styles.reviewWrapper}>
        <View style={styles.reviewItemWrapper}>
          <View style={styles.reviewerInfoWrapper}>
            <Image style={styles.imageAvatar} source={{ uri: reviewers?.image }} />

            <View style={styles.reviewerItemInfo}>
              <Text style={styles.textName}>{reviewers?.name}</Text>

              <View style={styles.reviewTime}>
                <Image style={styles.imageClock} source={IMAGES.iconClock} />
                <Text style={styles.textDate}>{reviewers?.date}</Text>
              </View>
            </View>
          </View>
          {/* End of review Info */}

          <View style={styles.reviewItemRating}>
            <View style={styles.ratingTextWrapper}>
              {Boolean(rating) && <Text style={styles.ratingTextPoint}>{rating}</Text>}
              <Text style={styles.ratingText}>rating</Text>
            </View>

            <Image style={styles.imageStar} source={IMAGES.ratingStart} />
          </View>
          {/* End of review Rating */}
        </View>

        <View style={styles.reviewItemContent}>
          {Boolean(comment) && (
            <MoreLessText styleShowMoreText={styles.textContent} numberOfLines={2}>
              {comment}
            </MoreLessText>
          )}
        </View>
        {/* End of review Content */}
      </View>
    </View>
  );
};

export default memo(Reviews, isEqual);
