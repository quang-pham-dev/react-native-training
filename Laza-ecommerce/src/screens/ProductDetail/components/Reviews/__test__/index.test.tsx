import React from 'react';
import { Image, Text } from 'react-native';

// LIBS
import renderer from 'react-test-renderer';

// Components
import Title from 'components/Title';
import Reviews from 'screens/ProductDetail/components/Reviews';

// Mock data
import { product } from '__mocks__/dataMock';

// API
import { productsService } from 'api/products.api';

describe('Product detail reviewer review', () => {
  const props = {
    comment: product.comment,
    rating: product.rating,
    name: product?.reviewers?.name,
    image: product?.reviewers?.image,
    date: product?.reviewers?.date,
  };

  const tree = renderer.create(<Reviews {...props} />);

  test('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  test('should render product reviews', async () => {
    let resProduct = await productsService.getProductById(product.id);

    const renderReviews = renderer.create(
      <Reviews
        comment={resProduct.data.comment}
        rating={resProduct.data.rating}
        reviewers={resProduct.data.reviewers}
      />,
    );

    expect(renderReviews).toBeTruthy();
  });
});
