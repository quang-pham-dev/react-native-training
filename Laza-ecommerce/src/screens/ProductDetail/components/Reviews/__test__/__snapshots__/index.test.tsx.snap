// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product detail reviewer review should render correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "textTransform": "none",
          },
          Object {
            "color": "#1D1E20",
            "fontFamily": "Inter_600SemiBold",
            "fontSize": 17,
            "lineHeight": 18,
          },
        ]
      }
    >
      Reviews
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "textTransform": "none",
          },
          Object {
            "color": "#8F959E",
            "fontFamily": "Inter_400Regular",
            "fontSize": 13,
            "lineHeight": 14,
          },
        ]
      }
    >
      View All
    </Text>
  </View>
  <View
    style={
      Object {
        "justifyContent": "space-between",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 15,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": 15,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginTop": 8,
              }
            }
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 11,
                  "width": 11,
                }
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1D1E20",
                "fontFamily": "Inter_500Medium",
                "fontSize": 15,
                "lineHeight": 16,
              }
            }
          >
            4.5
          </Text>
          <Text
            style={
              Object {
                "color": "#8F959E",
                "fontFamily": "Inter_400Regular",
                "fontSize": 11,
                "lineHeight": 12,
                "marginLeft": 5,
              }
            }
          >
            rating
          </Text>
        </View>
        <Image
          source={1}
          style={
            Object {
              "height": 13,
              "width": 57,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "marginTop": 10,
        }
      }
    >
      <Text
        onTextLayout={[Function]}
        style={
          Array [
            Object {
              "fontFamily": "Inter_600SemiBold",
              "fontSize": 15,
              "lineHeight": 22,
            },
            Object {
              "color": "#8F959E",
              "fontFamily": "Inter_400Regular",
              "fontSize": 15,
              "lineHeight": 22,
            },
          ]
        }
      >
        lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam,adipisicing elit. Quisquam, quia.
      </Text>
    </View>
  </View>
</View>
`;
